Core/Services/UserService.cs

```diff
@@ -37,5 +38,16 @@ public async Task<IEnumerable<SystemUserViewModel>> GetUsersViewModelAsync()
                 PhoneNumber = user.PhoneNumber
             }).ToListAsync();
         }
+
+        public async Task<IEnumerable<SystemUserViewModel>> Filter(string userName)
+        {
+            return await _context.SystemUsers.Where(user => user.Name.ToLower().Contains(userName.ToLower())).Select(user => new SystemUserViewModel
+            {
+                Name = user.Name,
+                Email = user.Email,
+                Id = user.Id,
+                PhoneNumber = user.PhoneNumber
+            }).ToListAsync();
```

Comment:
ToListAsync? Not FirstOrDefaultAsync? There should just be one hit.

Reply:
It searches for all users containing the search, not just for one.